src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -310,6 +323,42 @@ get_common_simd_info (MonoClass *vector_klass, MonoMethodSignature *csignature,
 	return TRUE;
 }
 
+static MonoType*
+get_vector_t_elem_type (MonoType *vector_type)
+{
+	MonoClass *klass;
+	MonoType *etype;
+
+	g_assert (vector_type->type == MONO_TYPE_GENERICINST);
+	klass = mono_class_from_mono_type_internal (vector_type);
+	g_assert (
```

Comment:
Should we also check the namespace?

Reply:
We aren't doing so in `simd-intrinsics.c`, notably, but we could if desired.

I believe the expectation is that we're already only hitting this path for `System.Runtime.Intrinsics` or `System.Numerics` due to the higher level functions that can dispatch to these code paths.